tick/optim/prox/src/prox.cpp

```diff
@@ -1,99 +1,181 @@
-//
-// Created by Martin Bompaire on 26/10/15.
-//
-
 #include "prox.h"
 
-Prox::Prox(double strength) {
-    has_range = false;
-    this->strength = strength;
+Prox::Prox(double strength,
+           bool positive) {
```

Comment:
Not saying it's very important here, but another time we should consider adding an enum instead of bool toggle: https://akrzemi1.wordpress.com/2017/02/16/toggles-in-functions/